### ISSUE TITLE
AddIn not loading

### ISSUE THREAD

**Issue by jnSWPC** [OfficeDev/PnP-OfficeAddins#79]
> Note: This repo is only for questions related to its samples. If you have questions about how to use office.js or the Office developer platform, please post your question on https://stackoverflow.com. Tag your question with office-js or outlook-web-addins

**Question**

_Ask your question here._
I cloned the repo and was able to successfully compile with start npm start.  Then when I do npm run sideload, excel launches but I see an "one or more add-in load failed" status pops up in the bottom but the buttons show up in the ribbon.. when i click on the buttons i see status show Conotoso Data importer is working on.. , but nothing happens.. Just wondering if I am missing some pre-req.. Appreciate help and guidance.

**Comment by jnSWPC** [comment]
the version of office i have is Version 2004 ( 12730.20270)

**Comment by aafvstam** [comment]
What Add-in are you referring to? The repo contains several add-ins.

**Comment by jnSWPC** [comment]
excel-shared-runtime-scenario

**Comment by aafvstam** [comment]
Usually this is related to the security. In my case, as I already had a valid localhost certificate, the sideload did not generate a new one and failed to load as well. I had to add the directory containing the manifest.xml (create a share to the folder containing the manifest first so you have a \\machine\sharename location to put in your trusted catalogs. 

![image](https://user-images.githubusercontent.com/5305682/82224285-f6b23280-9923-11ea-8d37-ddae1cd18244.png)

Now I was able to run the Add-in (several other issues, but I believe it is WIP)

**Comment by jnSWPC** [comment]
Thank you..Could you please share the Office version you have? 

This is the version I have
![Version](https://user-images.githubusercontent.com/22331494/82228593-ed5aa300-98ee-11ea-80e7-a7a9dfcabb3f.PNG)

**Comment by aafvstam** [comment]
At this time I'm on:
![image](https://user-images.githubusercontent.com/5305682/82259645-3b59c000-995c-11ea-91f1-5170648c4b4d.png)

But I'm pretty sure I was on your version when I tested it as I checked the version before that.

**Comment by jnSWPC** [comment]
I have created a share for the folder where manifest.xml resides
![image](https://user-images.githubusercontent.com/22331494/82351885-cd94af00-99c2-11ea-89bd-af94dd743348.png)

Also when i check for cert, i see the following
![image](https://user-images.githubusercontent.com/22331494/82352157-32500980-99c3-11ea-95e9-645d2583427a.png)

when i execute npm run sideload , excel launches and i see the following error
![image](https://user-images.githubusercontent.com/22331494/82352279-5e6b8a80-99c3-11ea-84be-60f793bdad84.png)

Appreciate help.

**Comment by aafvstam** [comment]
Did you also build and start the webservice (npm install, npm start) ? 

If so you should be able to start the app in your browser (check the port in the npm start output):

![image](https://user-images.githubusercontent.com/5305682/82362118-23da1000-9a0c-11ea-9d45-ab40c4c412e1.png)


As the addin is a two part process, the running webservice and the Excel addin reaching out to this webservice you need to have the webservice up and running before starting to sideload Excel. Sometimes it takes some time to load the webservice so wait until you are able to see the page as descibed next before you sideload the Add-in.

Now as this page is up you should be able to start and sideload the Addin by running "npm run sideload". This kicks in Excel, should also sideload the Add-in (if already certs were use the workaround I earlier mentioned)

![image](https://user-images.githubusercontent.com/5305682/82362406-9814b380-9a0c-11ea-81a8-9eb7aa71a4f2.png)

[Note that the readme mentions things like "start npm start" as they trigger it from the cmd prompt. In my case I trigger the commands from VSCode Terminal so no need to prefix with start there]

Hope this helps ...

**Comment by jnSWPC** [comment]
Appreciate your detailed response.. yes I had the service running.. I see this certificate error.. could this be contributing to the issue I am seeing 
![image](https://user-images.githubusercontent.com/22331494/82391742-af04d700-9a07-11ea-86c1-ed77d95f93bb.png)

**Comment by aafvstam** [comment]
Yes, this can be very much your issue. Is that what you see if you run the service in the browser? You can go into certmgr and remove the invalid certificate. After that "npm run sideload" most likely installs a new certificate...

**Comment by jnSWPC** [comment]
Thank you very much.. I cleared up my existing certs from cert mgr and then installed the root certificate from excelshare\certs ca.crt into trusted catalog.. started service, navigated to https://localhost://3000, and got cert error, installed the localhost cert into trusted catalog from within browser.  now when i did a sideload the add-in loaded..

**Comment by aafvstam** [comment]
Great, good to hear! Certificates always is a bit of a pain. Office Apps are extremely good protected to avoid malware issues and other security risks. So to start it can be painful but once it runs you are good to go 😎👍

**Comment by jnSWPC** [comment]
I had another follow up question.. I have a visual studio office add-in project.. I am trying to explore the same shared runtime preview feature. I have referenced the beta office.js and the type. However during compile VS has a build failure as follows.. Is there a schema url for preview i can use that has the RunTimes definition? 
![image](https://user-images.githubusercontent.com/22331494/82592337-5ef05680-9b66-11ea-816c-f82ec9c0acca.png)

![image](https://user-images.githubusercontent.com/22331494/82592292-4b44f000-9b66-11ea-8b90-5cb30ddc482c.png)

**Comment by zlatko-michailov** [comment]
Sorry, I didn't see that you have already gotten a solution to this issue.

As your new issue is unrelated to these samples, please file it at [StackOverflow](https://stackoverflow.com) with an `officejs` tag, so the team can close this thread.
